### ISSUE TITLE
Invalid date-time can be generated, intermittently

### ISSUE THREAD

**Issue by lissyx** [taskcluster/json-e#200]
STR:
 - be on halloween day off
 - submit taskcluster tasks

Expected:
 Works.

Actual:
 Fails with "InputValidationError"

This is the error:
```
{
  "code": "InputValidationError",
   ...
...
      "created": "2017-11-01T22:00:09",
...
}
```

As you can see, `created`, `deadline` and `expires` are all invalid because they lack the timezone info. It is generated here [https://github.com/mozilla/DeepSpeech/blob/12dee36a43fbbebcccf1a3c2cbc8946c54924f7d/taskcluster/linux-opt-base.tyml#L10-L15]:
```
created: { $fromNow: '0 sec' }
deadline: { $fromNow: '1 day' }
expires:
  $if: 'event.event == "push"'
  then: { $fromNow: '6 months' }
else: { $fromNow: '7 days' }
```

This template is being ingested by JSON-e Python module.

*[edit: removed literal newlines as they are confusing TaskWarrior]*

**Comment by jonasfj** [comment]
my suspicion is that the following magic doesn't work reliably:
https://github.com/taskcluster/json-e/blob/master/jsone/shared.py#L110-L114
(like maybe there is a magic time at which it doesn't work)

**Comment by lissyx** [comment]
```
>>> d2 = datetime.datetime(2017, 11, 1, 22, 0, 9, 0)
>>> stringDate(d2)
'2017-11-01T22:00:09'
>>> stringDate(d)
'2017-11-02T00:32:54.010Z'
>>> 
```

**Comment by lissyx** [comment]
and ...
```
>>> d3 = datetime.datetime(2017, 11, 1, 22, 0, 9, 1)
>>> stringDate(d3)
'2017-11-01T22:00:09.000Z'
>
```

**Comment by lissyx** [comment]
Looks like this is because of https://docs.python.org/3/library/datetime.html#datetime.datetime.isoformat: "Return a string representing the date and time in ISO 8601 format, YYYY-MM-DDTHH:MM:SS.mmmmmm or, if microsecond is 0, YYYY-MM-DDTHH:MM:SS"

**Comment by lissyx** [comment]
Fix could be as simple as adding `timespec='microseconds'`:
```
>>> str(datetime.datetime(2017, 11, 2, 0, 32, 54, 0).isoformat())
'2017-11-02T00:32:54'
>>> str(datetime.datetime(2017, 11, 2, 0, 32, 54, 0).isoformat(timespec='microseconds'))
'2017-11-02T00:32:54.000000'
>>> 
```

**Comment by lissyx** [comment]
FYI, since I filed this bug, we got something like 10 occurrences of this intermittence.

**Comment by djmitche** [comment]
Do you want to make a PR?  Looks like timespec='microseconds' would fix it?

**Comment by lissyx** [comment]
I can make it a PR if you think it's a good enough fix :)

**Comment by djmitche** [comment]
Fixed in v2.3.5.

**Comment by lissyx** [comment]
Looks like it's not just py2/py3 debate: `TypeError: 'timespec' is an invalid keyword argument for this function` https://tools.taskcluster.net/groups/PQXqB4WaQRSc1hi6ge15Kg/tasks/PQXqB4WaQRSc1hi6ge15Kg/runs/0/logs/public%2Flogs%2Flive_backing.log#L565

Yet this runs under Python 3.

**Comment by lissyx** [comment]
@djmitche It seems like I got tricked, `timespec` is only available on v3.6 as much as I could test, (3.5 from debian 9.2 does not have it).

**Comment by lissyx** [comment]
@djmitche I re-opened another PR relying on the `TypeError` since so far this would be more reliable ; I had a hard time finding proper documentation on which specific version does have that.

https://github.com/taskcluster/json-e/pull/207

**Comment by lissyx** [comment]
According to python/cpython@a2998a6#diff-25e2d173c84057d069b7890450714edd, it would be available since 3.5.2. Hence, I don't really see a reliable way to test that besides catching the exception :[

**Comment by jonasfj** [comment]
Maybe consider something more inline with:
https://github.com/taskcluster/taskcluster-client.py/blob/1eca2aa980170f9c9b46693abd5f097e0ef781a3/taskcluster/utils.py#L146-L157
